packages/react-charting/src/components/DonutChart/Arc/Arc.tsx

```diff
@@ -80,6 +111,59 @@ export class Arc extends React.Component<IArcProps, IArcState> {
     wrapTextInsideDonut(classNames.insideDonutString, this.props.innerRadius! * 2 - TEXT_PADDING);
   }
 
+  private _getTruncatedText(text: string, maxWidth: number): string {
+    const words = text.split(/\s+/).reverse();
+    let word: string = '';
+    const line: string[] = [];
+    let truncatedText = text;
+    const tspan = d3Select('#Donut_center_text').text(null).append('tspan');
+    let ellipsisLength = 0;
+
+    if (tspan.node() !== null && tspan.node() !== undefined) {
+      // Determine the ellipsis length for word truncation.
+      tspan.text('...');
+      ellipsisLength = tspan.node()!.getComputedTextLength();
+      tspan.text(null);
+      truncatedText = '';
+
+      while ((word = words.pop()!)) {
+        line.push(word);
+        tspan.text(line.join(' ') + ' ');
```

Comment:
You are truncating here without wrapping. First try to solve via wrapping and then by truncation. And lets discuss what all possible scenarios are there.

Reply:
Made the change to wrap first and then truncate. Added 7 corner cases too that were handled.

Reply:
In case 5 and case 6, you mentioned you are truncating vertically and horizontally, but you are truncating it only once. Can you clarify on this.

Reply:
Atishay, this change is now outdated and I have moved it to the utilities.ts in wrapTextInsideDonut(). After wrapping, if the text exceeds either horizontally (exceeds the central width) or vertically (if it goes beyond the inner circle vertically), then we are adding the ellipsis and breaking. If we want to continue truncation even after the text is truncated one way, we will need to add ellipsis both horizontally and vertically might not be a good experience.

Reply:
My comment is on the description that you have provided for case 5 and case 6. Here you are truncating only once and not both ways. Can you update the description.

Reply:
Updated description.